xds/internal/client/xds.go

```diff
@@ -620,3 +728,27 @@ func parseEDSRespProto(m *v3endpointpb.ClusterLoadAssignment) (EndpointsUpdate,
 	}
 	return ret, nil
 }
+
+func combineErrors(rType string, topLevelErrors []error, perResourceErrors map[string]error) error {
+	var errStrB strings.Builder
+	errStrB.WriteString(fmt.Sprintf("error parsing %q response: ", rType))
+	if len(topLevelErrors) > 0 {
+		errStrB.WriteString("top level errors: ")
+		for i, err := range topLevelErrors {
+			if i != 0 {
+				errStrB.WriteString("; ")
```

Comment:
Are these errors readable? Without any newlines, and just `;` as separators, I'm not very sure how readable these errors will be in the logs.

Reply:
I don't like newlines in logs and errors, but maybe that's a good idea in this situation.